### ISSUE TITLE
Error raised in Monthview

### ISSUE THREAD

**Issue by lmigula** [twinssbc/Ionic-Calendar#90]
Hi,
I stumbled over a little bug in the month view.

At line 627 of the calendar-tpls.js the code will run following error "TypeError: Cannot set property 'hasEvent' of undefined" when `timeDifferenceEnd - eps` is greater than 42.
The monthview works fine when making sure that timeDifference isn't greater than 42 before getting to the while loop
`if(timeDifferenceEnd > 42){
     timeDifferenceEnd = 42;
}`
or check that `dates[index]` exists in the loop before setting `dates[index].hasEvent = true;`.

The error can sometimes be reproduced in the [demo](http://twinssbc.github.io/Ionic-Calendar/demo/#/tab/home) when events are loaded.

**Comment by twinssbc** [comment]
@Imigula
I'm still trying to reproduce it and so far no luck.
I can't understand how timeDifferenceEnd can be larger than 42.
As you can see the below code.
timeDifferenceEnd won't be larger than (et-st)/oneDay.
et and st are the date range of the current monthview, so it will be always 42 days, unless somewhere st and et are calculated wrong.
Any chance you know the error will happen on a particular month?

```
                        var timeDifferenceEnd;
                        if (eventEndTime >= et) {
                            timeDifferenceEnd = (et - st) / oneDay;
                        } else {
                            timeDifferenceEnd = (eventEndTime - st) / oneDay;
                        }

                        var index = Math.floor(timeDifferenceStart);
                        var eventSet;
                        while (index < timeDifferenceEnd - eps) {
                            dates[index].hasEvent = true;

```

**Comment by lmigula** [comment]
Maybe it's something with the locale timezone.
I can reproduce the error in the october view in the demo that ends on the 5th of november.
The values for et, st,.. are the following for example:
`et: Sat, 05 Nov 2016 23:00:00 GMT
st: Sat, 24 Sep 2016 22:00:00 GMT
timeDifferenceEnd: 42.041666666666664
eventStartTime: Sat, 05 Nov 2016 12:35:00 GMT
eventEndTime: Sun, 06 Nov 2016 12:44:00 GMT
event: {"title":"Event - 41","startTime":"2016-11-05T12:35:00.000Z","endTime":"2016-11-06T12:44:00.000Z","allDay":false}`
I stumbeld upon it when i had I my data an event that started in October and ended on the 10th of november.

**Comment by twinssbc** [comment]
@lmigula Yes, it's the time zone issue.
Let's say Sep 30 is the end of DST.
If I calculate like below, it will return 25 hours.
But I only have 24 cells for one day, so it will cause setting value out of range.
The same issue applies to the month view that you met.

Since I'm not in DST time zone, I'm not quite familiar about how calendar works in DST time zone.
May I know how the time related website/app usually works in your life?
How to put things happen in 25 hours into 24 hours?
```
var st = new Date(2016, 08, 30)
var et = new Date(2016, 09,01)
(et-st)/3600000 = 25
```

**Comment by twinssbc** [comment]
@lmigula I have fixed the bug and released in version 0.3.2.
Feel free to have a try. Thanks.

**Comment by lmigula** [comment]
@twinssbc Sorry for my late answer. I totally forgot about about the end of DST as a cause. 
Since the end of DST means that the clock will be set back from 3am to 2am again and basically everyone hopes not to set a calendar entry at that time ;) Or as workaround it will be with a hint if it's 2am DST or 2am non DST.
As far as I can see your change works. I replaced it with my workaround and I didn't get an error.

Thanks :)
